assets/src/edit-story/components/canvas/page.js

```diff
@@ -41,6 +41,12 @@ function Page() {
 	const [ targetEl, setTargetEl ] = useState( null );
 	const [ pushEvent, setPushEvent ] = useState( null );
 
+	const targetRefs = useRef( [] );
+	useEffect( () => {
+		// Update the size of targetRefs when the selected elements changes.
+		targetRefs.current = targetRefs.current.slice( 0, selectedElements.length );
+	}, [ selectedElements ] );
```

Comment:
If there's a more reasonable way for storing the refs of the selected elements, please do share.

Reply:
Yes, that's the goal. Do you mean storing the node within the `selectedElements`? (If I understood correctly then) AFAIK that's not really recommended, the state should store serializable application state while DOM node is a visual representation of that instead. @barklund perhaps can chime in here?